### PR TITLE
Week 3: Comments

### PR DESCRIPTION
**Pull request by alisa-lin**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,92 @@
...
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, name, text, timestamp);
+      comments.add(comment);
+      numComments++;
+    }
+
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
```
Comment (pakraider) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You don't need the semicolon in "application/json"

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (MarsWang42) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:33]: This should actually be a DELETE request, is it the tutorial that asked you to do a post here?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -11,3 +11,43 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/** Fetches existing comments and updates UI. */
+function getComments() {
+    document.getElementById('comments').innerHTML = "";
+    var maxComments = document.getElementById("maxComments").value;
+    fetch('/data?max=' + maxComments).then(response => response.json()).then(comments => {
```
Comment (MarsWang42) [inline portfolio/src/main/webapp/script.js]: You can try use JavaScript template syntax here.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 2 above)
Comment (alisa-lin) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:33]: yeah, they asked us to doPost

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (alisa-lin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: fixed!

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (alisa-lin) [reply portfolio/src/main/webapp/script.js]: Thanks!

File: portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java (hunk 2 above)
Comment (MarsWang42) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:33]: Got it! Just keep in mind in the future try use a DELETE http request for deleting :)

**Review by MarsWang42 — APPROVED** [review]

**Review by pakraider — APPROVED** [review]

**Pull request merged**